src/app/DataVersionFilter.h

```diff
@@ -37,9 +35,10 @@ struct DataVersionFilter
         return (mEndpointId != kInvalidEndpointId) && (mClusterId != kInvalidClusterId) && (mDataVersion.HasValue());
     }
 
-    EndpointId mEndpointId = kInvalidEndpointId;
-    ClusterId mClusterId   = kInvalidClusterId;
-    Optional<DataVersion> mDataVersion;
+    ClusterId mClusterId = kInvalidClusterId;        // uint32
+    Optional<DataVersion> mDataVersion;              // uint32
+    DataVersionFilter * mpNext = nullptr;            // pointer width (32/64 bits)
```

Comment:
Again, this interacts weirdly with the places where we are creating arrays of these.

We should really think through how that should be handled for all these classes.

Reply:
template <typename T>
struct ClusterInfo
{
    T mValue;
    ClusterInfo * mpNext = nullptr;
};

just see your comments for 16467, how about creating one template clusterInfo class
where T could be AttributePathParams, EventPathParams, and DataVersionFilter
Although clusterInfo is still triple-purpose, its memory has been reduced a little bit based upon T

Reply:
offline chat with boris, would go the above way.

Reply:
updated, thanks